src/Bridge.ts

```diff
@@ -679,10 +688,29 @@ export class Bridge {
         await retry(() => this.as.botIntent.joinRoom(roomId), 5);
         if (event.content.is_direct) {
             const room = await this.setupAdminRoom(roomId, {admin_user: event.sender}, NotifFilter.getDefaultContent());
-            await this.as.botIntent.underlyingClient.setRoomAccountData(
+            await this.as.botClient.setRoomAccountData(
                 BRIDGE_ROOM_TYPE, roomId, room.accountData,
             );
         }
+
+        if (this.connectionManager?.isRoomConnected(roomId)) {
+            // Room has connections, don't setup a wizard.
+            return;
+        }
+
+        try {
+            // Otherwise it's a new room
+            if (this.config.widgets?.roomSetupWidget?.addOnInvite) {
+                if (await this.as.botClient.userHasPowerLevelFor(this.as.botUserId, roomId, "im.vector.modular.widgets", true) === false) {
+                    await this.as.botIntent.sendText(roomId, "Hello! To setup new integrations in this room, please promote me to a Moderator/Admin");
+                } else {
+                    // Setup the widget
+                    await SetupWidget.SetupRoomConfigWidget(roomId, this.as.botIntent, this.config.widgets);
+                }
+            }
+        } catch (ex) {
+            log.warn(`Failed to setup new widget for room`, ex);
```

Comment:
Pretty much all catch statements in this codebase use `ex` for Exceptions.

Reply:
Oh dear, I should've been more specific, sorry for that :D­

What I meant is: since we're handling an exceptional situation, shouldn't this be `log.error` rather than `log.warn`?

Reply:
I think my brain jumped a line, fixed :)